src/main/kotlin/fi/metatavu/lipsanen/projects/milestones/tasks/TaskController.kt

```diff
@@ -80,6 +81,13 @@ class TaskController {
             endDate = task.endDate,
             milestone = milestone,
             status = TaskStatus.NOT_STARTED,
+                // Todo: match types below
+            assignees = emptyList(),
+            userRole = task.userRole ?: UserRole.USER,
+            estimatedDuration = task.estimatedDuration ?: "",
+            estimatedReadiness = task.estimatedReadiness ?: "",
+                // Todo: match types below
+            attachments = emptyList(),
```

Comment:
Same logic as with TaskAssigneeEntity here, you will have to use TaskAttachmentRepository to save it. 
Note that both should happen after the task itself it saved so that they can get reference to the valid object, e.g.
create task
create attachment for the task
create assignee for the task